ksql-engine/src/main/java/io/confluent/ksql/topic/TopicDeleteInjector.java

```diff
@@ -123,6 +125,22 @@ public TopicDeleteInjector(
     return statement.withStatement(withoutDeleteText, withoutDelete);
   }
 
+  private void checkTopicError(final Exception error, final String sourceName) {
+    if (!(error instanceof KafkaDeleteTopicsException
+            && ((KafkaDeleteTopicsException) error).getExceptionList().get(0).getValue()
+            instanceof UnknownTopicOrPartitionException)) {
+      throw new KsqlException("Could not delete the corresponding kafka topic: "
+              + sourceName, error);
+    }
+  }
+
+  private void checkSchemaError(final Exception error, final String sourceName) {
+    if (!error.getMessage().contains("Subject not found")) {
```

Comment:
What kind of exception implementation do you actually get thrown here? There should be a better way to check for a missing subject.

Reply:
https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/rest/exceptions/RestClientException.java

https://docs.confluent.io/current/schema-registry/develop/api.html

Some other exceptions besides RestClient can occur such as when the schema registry is down, so I first check to make sure the exception is a RestClient one before checking the error code that corresponds to a missing schema.